src/anemoi/inference/forcings.py

```diff
@@ -132,29 +134,14 @@ def __repr__(self) -> str:
         """Return a string representation of the ComputedForcings object."""
         return f"{self.__class__.__name__}({self.variables})"
 
-    def load_forcings_array(self, dates: List[Date], current_state: State) -> FloatArray:
-        """Load the computed forcings for the given dates.
-
-        Parameters
-        ----------
-        dates : List[Date]
-            The list of dates for which to load the forcings.
-        current_state : State
-            The current state of the model.
+    def load_forcings_array(self, date: Date, *, initial: bool) -> FloatArray:
 
-        Returns
-        -------
-        FloatArray
-            The loaded forcings as a numpy array.
-        """
         LOG.debug("Adding dynamic forcings %s", self.variables)
 
-        if not isinstance(dates, (list, tuple)):
-            dates = [dates]
-
+        dates = [date + h for h in self.checkpoint.lagged] if initial else [date]
         source = UnstructuredGridFieldList.from_values(
-            latitudes=current_state["latitudes"],
-            longitudes=current_state["longitudes"],
+            latitudes=self.checkpoint.latitudes,
+            longitudes=self.checkpoint.longitudes,
```

Comment:
Having this hardcoded here is a problem. We need the logic of first trying to take the lat/lon from the input fields, and then using the checkpoint as a fallback. That seems to have disappeared.